compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -131,6 +131,9 @@ impl<K: Hash + Eq, V> interpret::AllocMap<K, V> for FxHashMap<K, V> {
 
     #[inline(always)]
     fn filter_map_collect<T>(&self, mut f: impl FnMut(&K, &V) -> Option<T>) -> Vec<T> {
```

Comment:
This site depends on the callers, but the only caller is the leak checker. So we could either make the function name end with like `_only_use_in_miri` or just fix it and let miri take the perf hit

Reply:
I think the perf hit should be tiny for the usual case where programs have freed most of their memory when they terminate.

Reply:
But there's another perf hit somewhere, right? After all we are seeing a regression in the benchmarks.